### DISCUSSION TITLE
Delete alternative routes

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#208] [Q&A] [answered]
when I set the Brouter offline route calculation alternatives to 2 and select the 2nd calculated route, how can I then hide the other route, after having selected one finally?

when I for example insert another wp then the selected route get's changed again

**Comment by devemux86** [comment]
With alternative routes, we can select the active route (tap on the route).
Then the selected active route is used in route planning and navigation.

When we set another waypoint on any of the alternative routes,
if alternative routes is still enabled, then 2+ routes are calculated again.

Or do you ask to have an option to delete one of the alternative routes on the map?

**Comment by Ciutnik** [comment]
By placing an intermediate point on the SELECTED route, it changes ALL alternative routes. While in the case of via-points such behavior is expected, in the case of shaping points it would be good if it concerned only the SELECTED route. That's my point by the way.

**Reply by devemux86** [reply]
> By placing an intermediate point on the SELECTED route, it changes ALL alternative routes.

An alternative route means another path that passes through the same points.
Changing the waypoints makes the route + alternatives to be calculated again.

BRouter is the only one that calculates alternate routes with >2 waypoints.
All other routers usually calculate alternative routes when waypoints = 2.

**Reply by Ciutnik** [reply]
Right. BRouter doesn't seem to distinguish via-points from shaping points. I forgot about that 🙂

**Reply by devemux86** [reply]
> BRouter doesn't seem to distinguish via-points from shaping points.

There is work planned for this. :slightly_smiling_face: 
- https://github.com/abrensch/brouter/issues/543

**Comment by DolomitesTom** [comment]
> Or do you ask to have an option to delete one of the alternative routes on the map?

Yes I guess this would help. Then you could refine further by calculating again two alternatives, delete one and further refine the selection.

**Reply by devemux86** [reply]
@DolomitesTom 

I have put it in my implementation list. :slightly_smiling_face:

**Comment by devemux86** [comment]
Next Cruiser will have an option in the route (long press) menu to delete an alternative route.

The action will be included in the routing history, so you can revert it using the [Undo button](https://github.com/devemux86/cruiser/discussions/199).

**Reply by devemux86** [reply, marked as answer]
Delete of alternative routes is available in [Cruiser 3.0.24](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

Please see above for details: https://github.com/devemux86/cruiser/discussions/208#discussioncomment-6162496

**Reply by DolomitesTom** [reply]
It works, but it could be improved. When you tap on the alternative, the whole route selection changes. But maybe you only want to keep a certain section between 2 waypoints. Would it be possible to long press on a section and the select an option "keep this section"? The the original route stays like it is an only the selected section will be added to the planned active route? Know what I mean?

**Reply by devemux86** [reply]
Split and merge routes is a complicated process that currently exists in Cruiser navigation rerouting.

I could see such feature in route planning as part of:
- https://github.com/devemux86/cruiser/discussions/142

**Reply by DolomitesTom** [reply]
fine, has low priority
